### DISCUSSION TITLE
How can I add translation?

### DISCUSSION THREAD

**Discussion by MMate2007** [romcal/romcal#402] [Q&A] [answered]
Hello Everybody,

I would like to add Hungarian translation to the project. How can I do that?

Thank you for your help!

Máté

**Comment by tukusejssirs** [comment, marked as answer]
@MMate2007, simply localise [`lib/locales/en.ts`](https://github.com/romcal/romcal/blob/dev/lib/locales/en.ts) (copy it to `hu.ts`), update the array of locales in [`lib/locales/index.ts`](https://github.com/romcal/romcal/blob/dev/lib/locales/index.ts).

Note that you don’t have to localise all the strings, although that would be awesome. Ideally, you should localise all strings used in the Hungarian Proper (which might require some proofing, in a separate PR though), which is defined in [`lib/particular-calendars/hungary.ts`](https://github.com/romcal/romcal/blob/dev/lib/particular-calendars/hungary.ts).

Anyway, a WIP PR is a good start in order to let us review your partial work and to make sure you don’t do some unnecessary work.

Feel free to ask any questions, esp when you are not sure if something is right or wrong (I localised, proofed and fixed English, Czech, Latin and Slovak locales). Try to adhere to official sources (esp the latest Hungarian translation of the Roman Missal, Ordos / liturgical calendars, other liturgical books) unless a particular celebration name (or a personal name) is not found it it, in which case try to find the best translation available online. At all times, try to be consistent.

Anyway, we are really grateful for any contribution! :pray:

**Comment by MMate2007** [comment]
And after some changes I can create a PR to this repository to the dev branch?

**Reply by tukusejssirs** [reply]
Yes, to the `dev` branch and mark it as WIP (draft) until you consider it finished.

And I’d like to ask you to reply to a message you are reacting to instead of creating a new message (just like I do it here). Failing to do so, no harm is done (I too sometimes choose the wrong text box), however, it might become a mess when we have a conversation on one topic in multiple places. :wink:

**Reply by MMate2007** [reply]
OK, thank you for your help!

**Reply by MMate2007** [reply]
Can I use a dictionary for this work?

**Reply by tukusejssirs** [reply]
You should, but use the translation (in romcal) with grain of salt and only when there is no official translation of the string, e.g. strings found in [_Direktórium_](https://www.liturgia.hu/direktorium/) should _not_ be (re-) translated.

Anyway, if you have access to the Hungarian translation of the Roman Missal (any and all editions since the Second Vatican Council), I’d be interested in the General Calendar (Proper) for Hungary (or the Hungarian Dioceses), which is always printed after the intro (usually around page 100) which is a list of celebrations inscribed in Hungarian calendar divided per month. It’d be awesome if you could scan (or simply take a picture) of all the months including the front cover, title page, title page verso (the page following the title page) and colophon (if exists; last page of the book, excluding indices) which I’d use for citation and reference purposes. I’d be also interested in Calendar Propers for any and all Hungarian dioceses, including decrees that change the celebrations inscribed in them (both national and diocesan Propers).

**Reply by tukusejssirs** [reply]
@MMate2007, may I suggest to comment out every string that is not yet localised? It is not required, however, in my experience, it helps a lot when looking for unlocalised strings.

Also moving all values to the same line on which is its key helps sorting the file. Then, I usually always move the unlocalised string to the end of `names` object.
